tests/api/test_api.py

```diff
@@ -88,23 +92,40 @@ def get_api_test_info(path):
 
         output_modifier = globals()[modifier_name]
 
-    return import_module_name, arguments, data['output'], output_modifier
+    return import_module_name, arguments, data['output'], is_error, output_modifier
 
 def _get_api_test_method(path):
-    import_module_name, arguments, expected, output_modifier = get_api_test_info(path)
+    import_module_name, arguments, expected, is_error, output_modifier = get_api_test_info(path)
 
     def __method(self):
         api_module = importlib.import_module(import_module_name)
 
         self._next_response_queue.put(expected)
-        actual = api_module.send(arguments)
+        try:
+            actual = api_module.send(arguments)
+        except Exception as ex:
+            if (not is_error):
+                raise ex
+
+            message = _unpack_expected_error_message(expected)
+            assert (message == str(ex))
```

Comment:
NOPE!

As a rule, we do not use `assert()` in tests.
`assert()` is for normal code, not test code.
We use the `TestCase` [assert methods](https://docs.python.org/3/library/unittest.html#assert-methods), but NEVER `assert()`.

Make sure to note what happens with this **method** this line is in,
i.e., what class is it attached to.

This is subtle but very important.
Really take time and look over everything carefully.